### DISCUSSION TITLE
VS Code - Please list all Codespaces/worktrees in "Remote Explorer" tab even when connected to a Codespace

### DISCUSSION THREAD

**Discussion by markfields** [community/community#33979] [Codespaces]
In a new VS Code window, the "Remote Explorer" window shows all codespaces and even shows each one's worktrees! So great! But once I connect to a Codespace, there's only one Heading on the "Remote Explorer" tab, which shows stats/status for the connected Codespace.  Please also show the same info as when I'm not connected, so I can easily switch!

**Comment by bjbennet** [comment]
Hi @markfields, thanks for your suggestion. It was an intentional decision to only show the currently connected codespaces in the Remote Explorer when inside of that codespace. This is a primary due to the access tokens that are minted when connected to a particular codespaces. These tokens do not have access to other codespaces.

**Reply by markfields** [reply]
Ok. I was hoping that whatever was displayed when not connected would still be accessible, and could go in a separate list, could be collapsed by default.  

User scenario is having a single consistent place to view all my worktrees across codespaces. Workaround is to open a new VS Code window which is not too bad.

**Reply by markfields** [reply]
Update: It looks like the Worktrees view in that view when disconnected needs some work anyway, it seems to have gotten out of date compared to the GitLens Worktree view when connected to each workspace.  I think this scenario is really compelling though - to be able to view all active worktrees across remotes and easily navigate between them.

[edit: I may just be confused, it may not be wrong]
